cmd/cdi-cloner/cloner_startup.sh

```diff
@@ -16,84 +16,30 @@
+    echo "VOLUME_MODE missing" 1>&2
     exit 1
 fi
 
-obj="$1"      # source|target
-rand_dir="$2" # part of socket path
-volumeMode="$3" # FS|Block
-
-pipe_dir="/tmp/clone/socket/$rand_dir/pipe"
-image_dir="/tmp/clone/image"
-if [ "$volumeMode" == "block" ]; then
-    image_dir="/dev/blockDevice"
+if [[ -z "$MOUNT_POINT" ]]; then
+    echo "MOUNT_POINT missing" 1>&2
+    exit 1
 fi
-echo "image_dir is" $image_dir
 
-retries=0
-max_retries=20
-sleep_time=3
+echo "VOLUME_MODE=$VOLUME_MODE"
+echo "MOUNT_POINT=$MOUNT_POINT"
 
-if [ "$obj" == "source" ]; then
-    echo "cloner: Starting clone source"
-    echo "cloner: creating fifo pipe"
-    mkfifo $pipe_dir
-    
-    echo $volumeMode
-    if [ "$volumeMode" == "FS" ]; then
-        echo "cloner: creating tarball of the image and redirecting it to $pipe_dir"
-        pushd $image_dir
-        #figure out the size of content in the directory
-        size=$(du -sb . | cut -f1)
-        echo size is $size
-        #Write the size to the pipe so the other end can read it.
-        printf "%016x" $size >$pipe_dir
-        tar cv --sparse . >$pipe_dir
-        popd
-    elif [ "$volumeMode" == "block" ]; then
-        #figure out the size of the block device
-        size=$(lsblk -n -b -o SIZE $image_dir) 
-        echo size is $size
-        #Write the size to the pipe so the other end can read it.
-        printf "%016x" $size >$pipe_dir
-        echo "cloner: writing the image to $pipe_dir"
-	    dd if=$image_dir bs=64K of=$pipe_dir
-    fi
-    echo "cloner: finished writing image to $pipe_dir"
-    exit 0
-fi
+if [ "$VOLUME_MODE" == "block" ]; then
+    UPLOAD_BYTES=$(lsblk -n -b -o SIZE $MOUNT_POINT)
+    echo "UPLOAD_BYTES=$UPLOAD_BYTES"
 
-if [ "$obj" == "target" ]; then
-    echo "cloner: Starting clone target"
-    while true; do
-        echo "cloner: check if the fifo pipe was created by the cloning source pod"
-        if [ -e "$pipe_dir" ]; then
-            if [ "$volumeMode" != "block" ]; then
-                pushd $image_dir
-            fi
-            echo "cloner: extract the image from $pipe_dir into $image_dir directory"
-            /usr/bin/cdi-cloner -pipedir $pipe_dir -alsologtostderr -v=3
-            if [ "$volumeMode" != "block" ]; then
-                popd
-            fi
-        	if [ "$?" != "0" ]; then
-        		echo "cloner: failed with exit code $?"
-        		exit 1
-        	fi
-            echo "cloner: finished cloning image from $pipe_dir to $image_dir"
-            exit 0
-        fi
-        if ((retries == max_retries)); then
-            echo "cloner: failed after $retries retries to clone image"
-            exit 1
-        fi
-        echo "cloner: $retries: fifo pipe has not been created by the source pod. Waiting $sleep_time seconds before checking again..."
-        sleep $sleep_time
-        let retries+=1
-    done
-fi
+    /usr/bin/cdi-cloner -v=3 -alsologtostderr -content_type blockdevice-clone -upload_bytes $UPLOAD_BYTES < $MOUNT_POINT
```

Comment:
Just curious, is there a way to set this later on ? I know we can set the cdi-operator VERBOSITY using oc patch, not sure if it would apply here as well.

Reply:
These pods are created on demand, but right here they are hard coded. I would expect verbosity to be en environment variable passed in using the pod spec. My comment relates to the hard coded verbosity.

Reply:
One more question: I see '-alsologtostderr' which reminds me if we are in compliance with 'mustgather' for the logs from network cloning ?

Reply:
Our must gather script basically does a kubectl logs <podname> to gather the logs. I forget what the reason for the -alsologstderr flag is. But all the other pods do something similar.

Reply:
I think verbosity 3 is fine for these transient pods.  Also, clone source pod is not particularly chatty.

Reply:
The verbosity is just for the source pod right? If so I am fine with hard coded 3.